tinygrad/runtime/ops_clang.py

```diff
@@ -12,7 +12,7 @@ def __init__(self, cachekey="compile_clang", args:Optional[List[str]]=None):
   def compile(self, src:str) -> bytes:
     # TODO: remove file write. sadly clang doesn't like the use of /dev/stdout here
     with tempfile.NamedTemporaryFile(delete=True) as output_file:
-      subprocess.check_output(['clang', '-shared', *self.args, '-O2', '-Wall', '-Werror', '-x', 'c', '-fPIC', '-ffreestanding', '-nostdlib',
+      subprocess.check_output(['clang', '-shared', *self.args, '-O2', '-Wall', '-Werror', '-x', 'c', '-fPIC', '-ffreestanding', '-nostdlib', '-lsystem',
```

Comment:
Why do you need this?

Reply:
I ran this command on a M2 Mac 

```
DEBUG=5 DISABLE_COMPILER_CACHE=1 CLANG=1 python3 examples/llama3.py --download_model --size 8B
```

It fails with:

```
ld: dynamic executables or dylibs must link with libSystem.dylib for architecture arm64
```

The clang command that gets executed:

```
['clang', '-shared', '-march=native', '-O2', '-Wall', '-Werror', '-x', 'c', '-fPIC', '-ffreestanding', '-nostdlib', '-', '-o', '/var/folders/0k/nj410z392tsg9484w4f976m80000gn/T/tmp8be37epu']
```

So, we are building a shared library (`-shared`) with no stdlib (`-nostdlib`). The error message says for arm64 this is not allowed, so I had to add the `-lsystem` flag (could've also gotten away with removing `-nostdlib`).

Some of the tests are failing because `-lsystem` is a MacOS thing, so let me guard the inclusion of the flag.